packages/financial-templates-lib/src/price-feed/FallBackPriceFeed.js

```diff
@@ -0,0 +1,99 @@
+const { PriceFeedInterface } = require("./PriceFeedInterface");
+
+// An implementation of PriceFeedInterface that provides an order pricefeeds to fall back to
+// if the higher-priority ones fail for any reason.
+class FallBackPriceFeed extends PriceFeedInterface {
+  /**
+   * @notice Constructs new FallBackPriceFeed.
+   * @param {List} orderedPriceFeeds an ordered list of priceFeeds to fallback through.
+   *      All elements must be of type PriceFeedInterface. Pricefeeds fall back from beginning of the array to the end.
+   *      Must be an array of at least one element.
+   */
+  constructor(orderedPriceFeeds) {
+    super();
+
+    if (orderedPriceFeeds.length === 0) {
+      throw new Error("FallBackPriceFeed cannot be constructed with no constituent price feeds.");
+    }
+
+    this.priceFeeds = orderedPriceFeeds;
+  }
+
+  // Return first successfully fetched price or throw an error if they all fail.
+  getCurrentPrice() {
+    for (let _priceFeed of this.priceFeeds) {
+      const price = _priceFeed.getCurrentPrice();
+      if (!price) continue;
+      return price;
+    }
+
+    // If no pricefeeds return successfully, indicate failure.
+    return null;
+  }
+
+  async getHistoricalPrice(time, verbose = false) {
+    // If failure to fetch any constituent historical prices, then throw
+    // array of errors.
+    let errors = [];
+    for (let _priceFeed of this.priceFeeds) {
+      try {
+        return await _priceFeed.getHistoricalPrice(time, verbose);
+      } catch (err) {
+        errors.push(err);
+        continue;
+      }
+    }
+
+    throw errors;
+  }
+
+  // Note: This method will fail if one of the pricefeeds has not implemented `getHistoricalPricePeriods`, which
+  // is basically every price feed except for the CryptoWatchPriceFeed.
+  getHistoricalPricePeriods() {
+    throw new Error("getHistoricalPricePeriods Unimplemented for FallBackPriceFeed");
+  }
+
+  // Gets the update time for the first pricefeed that successfully updated.
+  getLastUpdateTime() {
+    for (let _priceFeed of this.priceFeeds) {
+      const lastUpdateTime = _priceFeed.getLastUpdateTime();
+      if (!lastUpdateTime) continue;
+      return lastUpdateTime;
+    }
```

Comment:
Falling back on methods like this are a little weird. I'm not exactly sure how to handle it.

What about taking the latest of all successful return values? Not strongly opinionated, but this method seems to rely on the update having _never_ succeeded, which may not be true...? Really not sure :/

Reply:
Shouldn't this return the update time of the pricefeed that is used for the pricefeed, i.e. the first pricefeed to return successfully?

Reply:
Ok so as mentioned [here](https://github.com/UMAprotocol/protocol/pull/2718#discussion_r595626631), I'm leaning towards updating ALL pricefeeds (and catching + ignoring any errors), and then using the latest update time as the update time here.

Reply:
Yeah, I think that makes sense.

> Shouldn't this return the update time of the pricefeed that is used for the pricefeed, i.e. the first pricefeed to return successfully?
Ideally, yes, but I don't think it's true that this method errors when getCurrentPrice or getHistoricalPrice error, so you would effectively always get the first feed if that feed had ever succeeded.

Reply:
Yeah. Having the notion of "meta pricefeeds" such as Fallback and Medianizer makes the `update`-`getPrice` pattern inefficient, whereas a more robust `async getPrice` method would be better for the meta pricefeeds. This way the meta pricefeeds don't have to blindly update all constituent price feeds